### ISSUE TITLE
Create an array of a type generated by a custom type builder

### ISSUE THREAD

**Issue by Charly-xepelin** [jcteague/efate#134]
This is probably more a question than a request.

I have a type called `Rut`.

I created a generator for it so you can do:

```typescript
const personFixture = defineFixture<Person>((t) => {
    // other fields
    t.identifier.asRut();
}
```

How can I do if I need an array of `Rut`s?

Something similar to:

```typescript
const personFixture = defineFixture<Person>((t) => {
    // other fields
    t.identifiers.arrayOfFixture((x) => x.asRut(), 5);
}
```

**Comment by jcteague** [comment]
Yeah, this is definitely a weak spot.  With the current implementation you have two choices:
1. use the `as` operator, which is basically the built-in escape hatch and let's you do what ever you want.
2. create another extenstion `asArrayofRut()` which is going to cleanest approach if you have a lot of these.  You could use it to encapsulate the `as` operator.

Here is the [test](https://github.com/jcteague/efate/blob/ca39ce2803853e60542076b1837f57fed2422d9e/packages/efate/src/fixture.tests.ts#L294C4-L300C8) describing it's use.

I think in your case it probably look something like:
```TypeScript
const fixture = defineFixture<{ field: Rut[] }>((t) => t.field.as((inc) => t.asRut()));
```
